celavi/des.py

```diff
@@ -320,62 +323,97 @@ def pylca_interface_process(self, env):
             The SimPy environment this process belongs to.
         """
         while True:
-            print(f'{datetime.now()} In While loop pylca interface', flush=True)
-            time0 = time.time()
             yield env.timeout(self.timesteps_per_year)
-            print(str(time.time() - time0) + ' yield of env timeout pylca took these many seconds')
 
             annual_data_for_lci = []
             window_last_timestep = env.now
-            window_first_timestep = window_last_timestep - self.timesteps_per_year
-            time0 = time.time()
-            year = int(ceil(self.timesteps_to_years(env.now)))
+            window_first_timestep = env.now - self.timesteps_per_year
+
+            year = int(floor(self.timesteps_to_years(env.now)))
 
             for facility_name, facility in self.mass_facility_inventories.items():
                 process_name, facility_id = facility_name.split("_")
                 for material in self.possible_materials:
                     annual_transactions = facility.transaction_history.loc[
-                                          window_first_timestep:window_last_timestep + 1, material
+                                          window_first_timestep:window_last_timestep, material
                                           ]
-                    positive_annual_transactions = annual_transactions[annual_transactions > 0]
+                    sliced_info = facility.transaction_history.loc[
+                                          window_first_timestep:window_last_timestep, [material,'timestep']
+                                          ].reset_index()
+                    #Debugging file for mass transactions.
+                    """A problematic value occurs when in one sliced dataframe we have two years and the second year has a positive
+                    mass value. Due to that, calculating the proper year for the total mass sum becomes a little difficult
+                    For example - 
+                    A sliced dataframe contains 13 timesteps. From Jan 2021 to Jan 2022. Thus we can have two dataframe slices that have
+                    the time step corressponding to January of 2022 (one which ends in Jan 2022 and one which begins at Jan 2022)
+                    Manufacturing masses generally gets filled up in the first dataframe in the last time step. Thus Manufacturing mass years are
+                    displaced by -1 in the calculations We need to correct that. 
+                    if Manufacturing takes place in May 2021 and Jan 2022, its difficult to sum up the total mass for manufacturing
+                    for this particular dataframe slice because the year is spread out.
+                    Reducing the slice dataframe size results in all January activities and masses go missing. 
+                    """
+                    sliced_info['year'] = sliced_info['timestep']/self.timesteps_per_year+self.min_year                    
```

Comment:
Remove this lengthy comment - if there's something here that hasn't been resolved, create a separate issue for it with this content